clippy_lints/src/matches/single_match.rs

```diff
@@ -77,15 +77,31 @@ pub(crate) fn check<'tcx>(cx: &LateContext<'tcx>, ex: &'tcx Expr<'_>, arms: &'tc
                 }
             }
 
-            report_single_pattern(cx, ex, arm1, expr, els);
+            report_single_pattern(cx, ex, arm1, expr, els, contains_comments);
         }
     }
 }
 
-fn report_single_pattern(cx: &LateContext<'_>, ex: &Expr<'_>, arm: &Arm<'_>, expr: &Expr<'_>, els: Option<&Expr<'_>>) {
+fn report_single_pattern(
+    cx: &LateContext<'_>,
+    ex: &Expr<'_>,
+    arm: &Arm<'_>,
+    expr: &Expr<'_>,
+    els: Option<&Expr<'_>>,
+    contains_comments: bool,
+) {
     let lint = if els.is_some() { SINGLE_MATCH_ELSE } else { SINGLE_MATCH };
     let ctxt = expr.span.ctxt();
-    let mut app = Applicability::MachineApplicable;
+    let note = |diag: &mut Diag<'_, ()>| {
+        if contains_comments {
+            diag.note("you might want to preserve the comments from inside the `match`");
+        }
+    };
+    let mut app = if contains_comments {
+        Applicability::MaybeIncorrect
```

Comment:
It's still being applied with `MaybeIncorrect` apparently.

Reply:
The Clippy testsuite applies even `MaybeIncorrect` suggestions.  You can see in the `stderr` file that the note is present, indicating that the comments have been detected.

Maybe we should have, in the presence of `MaybeIncorrect` suggestions, both versions of the `fixed` file, one with only `MachineApplicable` and one with both `MachineApplicable` and `MaybeIncorrect`.

Reply:
Seems weird to apply a `MaybeIncorrect` suggestion.